rustls/src/client/hs.rs

```diff
@@ -242,16 +242,22 @@ fn emit_client_hello_for_retry(
     // should be unreachable thanks to config builder
     assert!(!supported_versions.is_empty());
 
+    // offer groups which are usable for any offered version
+    let offered_groups = config
+        .provider
+        .kx_groups
+        .iter()
+        .filter(|skxg| {
+            supported_versions
+                .iter()
+                .any(|v| skxg.usable_for_version(*v))
+        })
+        .map(|skxg| skxg.name())
```

Comment:
`filter_map()`?

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#/filter_map_bool_then clippy's on the ball!